README.md

```diff
@@ -0,0 +1,50 @@
+# hosts
+
+Allow user-defined hosts entries.
+
+`hosts` is a `LD_PRELOAD` library that overrides the fetching of hostnames in applications
+with functionality that fetches these entries from a user-defined hosts file instead of `/etc/hosts`.
```

Comment:
I don't think this is clear or accurate, hostnames aren't fetched, they are resolved to IP addresses. I think more precise description wouldn't hurt if you are doing the rewrite. The audience for this is rather deeply technical anyway.